### ISSUE TITLE
Issue with renderSection() on view layout, should be prevented from clearing data after display

### ISSUE THREAD

**Issue by addngr** [codeigniter4/CodeIgniter4#7069]
### PHP Version

7.4

### CodeIgniter4 Version

4.2.12

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

apache

### Database

MariaDB 10.2

### What happened?

It is not possible to reuse the string result of a View Layout $this->renderSection() due the function clear the data after display. 
I have a use case, Where, I need to re-use the section multiple times in View Layout.
E.g. 
```php
<!doctype html>
<html>
<head>
    <title><?= $this->renderSection('page_title') ?></title> <!------------- Work Fine ! -->
</head>
<body>
    <h1 class="heading"><?= $this->renderSection('page_title') ?></h1>  <!------------- Problem Here ! -->
    <?= $this->renderSection('content') ?>
</body>
</html>
```

### Steps to Reproduce

Use `renderSection()` multiple times for same variable in a View Layout.

### Expected Output

Expect to can save result of `renderSection()` and reuse.
```php
/**
    * Renders a section's contents.
    * @param bool $saveData If true, saves data for subsequent calls,
    *                             if false, cleans the data after displaying,
*/
public function renderSection(string $sectionName, bool $saveData = false)
{
    if (! isset($this->sections[$sectionName])) {
        echo '';

        return;
    }

    foreach ($this->sections[$sectionName] as $key => $contents) {
        echo $contents;
        if($saveData === false)
        {
             unset($this->sections[$sectionName][$key]);
        }
    }
}
```

**Comment by kenjis** [comment]
@lonnieezell Is this a bug?

**Comment by lonnieezell** [comment]
It's not a bug. I don't recall why it was unset, unless it was just to save memory that wasn't needed. 

I have no problem with a PR for this feature, though.

**Comment by kenjis** [comment]
@addngr Thank you for reporting.

This is not a bug, but a feature request.
If you send a PR for this, please send it to `4.4` branch.
See https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#branching

**Comment by kenjis** [comment]
Closed by #7126
